removed stuff we didn't end up doing from design doc (#40)
mlabs-haskell/DeNS: doc/Architecture.md

### Record Keys and Values

> Morally, the DeNS protocol is a cross-chain distributed Key-Value store, where the key component is a `(Name,Class)` pair managed by a Cardano smart-contract, and the value is an inline DNS record represented as a bytestring.

Reviewer: ```suggestion
Morally, the DeNS protocol is a cross-chain distributed Key-Value store, where the key component is a `(Name,Class)` pair managed by a Cardano smart-contract, and the value is an inline DNS record represented as a Plutus data type.
```

See here: https://github.com/mlabs-haskell/DeNS/blob/main/dens-validation/api/lbf/Dens.lbf#L43-L60

It's a bit more than just a ByteString.

        -- ^ Each set represents a Name Universe and must be tagged w/ the class ID corresponding to that universe
    }

> -- By convention, this should be a DNS resource record, but might be a pointer to offchain storage in future protocol iterations.
> -- `Nothing` indicates a deleted or reserved record

Reviewer: ```suggestion
-- By convention, this should be a DNS resource record, but might be a pointer to offchain storage in future protocol iterations.
```

> newtype DeNSValue = DeNSValue { densPointer :: Maybe ByteString }

Reviewer: ```suggestion
newtype DeNSValue = DeNSValue { densValue :: [DensRr] }

data DensRr = { ttl :: Integer, rData :: RData }

data RData 
    = A ByteString
    | AAAA ByteString
    | SOA ByteString

```

See here https://github.com/mlabs-haskell/DeNS/blob/main/dens-validation/api/lbf/Dens.lbf#L37-L60

### Records Validator

> The records validator serves as an access control list for the virtual DeNS database. When the offchain code reconstructs the database from Cardano keys and values (i.e. inline DNS-style resourcerecords, though in the future we may also support pointers to offchain records), it treats the records locked at the Records validator identified with a `ElementID` token as the source of authority. That is: Only records locked at the records validator are used to construct the database used in a DeNS resolver.

Reviewer: ```suggestion
The records validator serves as an access control list for the virtual DeNS database. When the offchain code reconstructs the database from Cardano keys and values (i.e. inline DNS-style resource records, though in the future we may also support pointers to offchain records), it treats the records locked at the Records validator identified with a `ElementID` token as the source of authority. That is: Only records locked at the records validator are used to construct the database used in a DeNS resolver.
```
